Meta/Lagom/Libraries/LibGUI/SDLServer.cpp

```diff
@@ -0,0 +1,186 @@
+            case SDL_WINDOWEVENT: {
+                auto window = Window::from_window_id(get_window_from_sdl_id(event.window.windowID));
+                switch (event.window.event) {
+                case SDL_WINDOWEVENT_RESIZED: {
+                    WindowServerConnection::the().window_resized(window->window_id(), Gfx::IntRect(0, 0, event.window.data1, event.window.data2));
+                    break;
+                }
+                }
+                break;
+            }
+            }
+        }
+
+        for (auto& entry : m_windows) {
+            auto window = Window::from_window_id(entry.key);
+            if (!window->is_visible())
+                continue;
+
+            Vector<Gfx::IntRect> rects = { window->rect() };
+            WindowServerConnection::the().paint(window->window_id(), window->size(), rects);
+
+            auto sdl_window = entry.value;
+            auto screen_surface = SDL_GetWindowSurface(sdl_window);
+
+            auto bitmap = window->back_bitmap();
+            if (!bitmap)
+                continue;
+
+            // FIXME: only do this if the bitmap is dirty
+            SDL_LockSurface(screen_surface);
+
+            // FIXME: This is a horrible way to do it, but dealing with SDL_BlitSurface is a pain future me
+            // will have to deal with :^)
+            for (int x = 0; x < bitmap->width(); x++) {
+                for (int y = 0; y < bitmap->height(); y++) {
+                    auto pixel = bitmap->get_pixel(x, y);
+
+                    Uint32* p_screen = (Uint32*)screen_surface->pixels;
+                    p_screen += y * screen_surface->w + x;
+                    *p_screen = SDL_MapRGBA(screen_surface->format, pixel.red(), pixel.green(), pixel.blue(), pixel.alpha());
+                }
+            }
+            SDL_UnlockSurface(screen_surface);
+
+            SDL_UpdateWindowSurface(sdl_window);
+        }
+    });
+    m_process_loop->start();
+}
+
+i32 SDLServer::get_window_from_sdl_id(u32 sdl_id)
+{
+    for (auto& entry : m_windows) {
+        auto sdl_window_id = SDL_GetWindowID(entry.value);
+
+        if (sdl_window_id == sdl_id)
+            return entry.key;
+    }
+
+    return -1;
+}
+
+void SDLServer::quit()
+{
+    dbgln("SDL: Quitting");
+    m_process_loop->stop();
+
+    SDL_Quit();
+    auto& loop = Core::EventLoop::current();
+    loop.quit(0);
+}
+
+void SDLServer::register_window(i32 window_id, SDL_Window* window)
+{
+    dbgln("SDL: Registering window with id {}", window_id);
+    m_windows.set(window_id, window);
+}
+
+void SDLServer::deregister_window(i32 window_id)
+{
+    dbgln("SDL: Removing window with id {}", window_id);
+    auto window = m_windows.get(window_id).value();
+    SDL_DestroyWindow(window);
+    m_windows.remove(window_id);
+}
+
+void SDLServer::set_window_title(i32 window_id, String title)
+{
+    SDL_SetWindowTitle(m_windows.get(window_id).value(), title.characters());
+}
+String SDLServer::get_window_title(i32 window_id)
```

Comment:
Please put new lines between function definitions (here and everywhere else)

Reply:
Totally forgot, I kept them together during development since it made it easier to make a mental map of associated functions. Will fix :^)